Models/Shape.cs

```diff
@@ -21,33 +49,86 @@ public class PolylineShape : ShapeBase
     public double StrokeThickness { get; set; } = 1.0;
     public List<Point> Points { get; set; } = new();
     
-    public override Control ToControl() 
+    protected override Rectangle CreateBoundingBoxVisual()
     {
-        return new Polyline
+        var rect = base.CreateBoundingBoxVisual();
+        rect.Stroke = Brushes.Red;
+        rect.StrokeThickness = 1;
+        return rect;
+    }
+    
+    public override Control ToControl()
+    {
+        var canvas = (Canvas)base.ToControl();
+
+        var polyline = new Polyline
         {
-                Points = new AvaloniaList<Point>(Points),
-                Stroke = Brushes.Black,
-                StrokeThickness = StrokeThickness
-       };
+            Points = new AvaloniaList<Point>(Points),
+            Stroke = Brushes.Black,
+            StrokeThickness = StrokeThickness
+        };
+
+        canvas.Children.Add(polyline);
+        return canvas;
+    }
+    public void AddPoint(Point next)
+    {
+        Points.Add(next);
+        double left   = Math.Min(BoundingBox.X, next.X);
+        double top    = Math.Min(BoundingBox.Y, next.Y);
+        double right  = Math.Max(BoundingBox.Right, next.X);
+        double bottom = Math.Max(BoundingBox.Bottom, next.Y);
+
+        BoundingBox = new Rect(left, top, right - left, bottom - top);
     }
 }
 
 public class GroupShape : ShapeBase
 {
     public List<ShapeBase> Children { get; set; } = new();
+    public void UpdateBoundingBox()
+    {
+        Rect? groupBox = null;
+
+        foreach (var child in Children)
+        {
+            if (groupBox is null)
+            {
+                groupBox = child.BoundingBox;
+            }
+            else
+            {
+                groupBox = Union(groupBox.Value, child.BoundingBox);
+            }
+        }
+        if (groupBox is not null)
+        {
+            BoundingBox = groupBox.Value;
+        }
+    }
     
     public override Control ToControl() 
     {
-        var panel = new Canvas();
+        UpdateBoundingBox();
```

Comment:
If the children also update their bounding boxes in `ToControl`, those changes won't be reflected here. It seems to me like you need to revert the ordering of operations, so that the child controls are created first, then the updated bounding box is calculated, and finally the parent control is built.